rxnetty-common/src/main/java/io/reactivex/netty/channel/BackpressureManagingHandler.java

```diff
@@ -374,33 +388,56 @@ public void channelWritabilityChanged(ChannelHandlerContext ctx) throws Exceptio
             super.channelWritabilityChanged(ctx);
         }
 
-        public void addSubscriber(final WriteStreamSubscriber streamSubscriber) {
-            streamSubscriber.add(Subscriptions.create(new Action0() {
+        public WriteStreamSubscriber newSubscriber(final ChannelHandlerContext ctx, ChannelPromise promise) {
+
+            recalculateMaxPerSubscriber();
+
+            final WriteStreamSubscriber sub = new WriteStreamSubscriber(ctx, promise, perSubscriberMaxRequest);
+            sub.add(Subscriptions.create(new Action0() {
                 @Override
                 public void call() {
                     /*Remove the subscriber on unsubscribe.*/
-                    subscribers.remove(streamSubscriber);
+                    subscribers.remove(sub);
+                    ctx.channel().eventLoop().execute(BytesWriteInterceptor.this);
                 }
             }));
 
-            subscribers.add(streamSubscriber);
+            subscribers.add(sub);
+            return sub;
         }
 
         /*Visible for testing*/List<WriteStreamSubscriber> getSubscribers() {
             return Collections.unmodifiableList(new ArrayList<>(subscribers));
         }
 
         private void requestMoreIfWritable(Channel channel) {
-            /**
-             * Predicting which subscriber is going to give the next item isn't possible, so all subscribers are
-             * requested an item. This means that we buffer a max of one item per subscriber.
-             */
+            assert channel.eventLoop().inEventLoop();
+
             for (WriteStreamSubscriber subscriber: subscribers) {
                 if (!subscriber.isUnsubscribed() && channel.isWritable()) {
-                    subscriber.requestMore(1);
+                    subscriber.requestMoreIfNeeded(perSubscriberMaxRequest);
                 }
             }
         }
+
+        @Override
+        public void run() {
+            recalculateMaxPerSubscriber();
+        }
+
+        /**
+         * Called from within the eventloop, whenever the subscriber queue is modified. This modifies the per subscriber
+         * request limit by equally distributing the demand. Minimum demand to any subscriber is 1.
+         */
+        private void recalculateMaxPerSubscriber() {
+            assert channel.eventLoop().inEventLoop();
+
+            int subCount = subscribers.size();
+            perSubscriberMaxRequest = 0 == subCount ? perSubscriberMaxRequest
+                                                    : perSubscriberMaxRequest * subCount / (subCount + 1);
+
+            perSubscriberMaxRequest = Math.max(1, perSubscriberMaxRequest);
```

Comment:
Yes it intentionally not being strict about exactly 64 as the demand is generated by the fact that the channel is ready to write more and 64 is more or less an artificial limit which is better than 1. Moreover, since the underlying buffer is in bytes and this limit is in terms of objects, there is less value in being precise.